automation/test.sh

```diff
@@ -170,143 +40,107 @@ for filename in dist/templates/*; do
-  _oc adm certificate approve $(_oc get csr -ocustom-columns=NAME:metadata.name --no-headers)
-
-  if [ $webhookUpdateRetries == 0 ];
-  then
-    echo Retry count for template validator ca bundle injection reached
-    exit 1
-  fi
-
-  webhookUpdateRetries=$((webhookUpdateRetries-1))
-
-  VALIDATOR_POD=$(_oc get pod -n $NAMESPACE -l kubevirt.io=virt-template-validator -o json | jq -r .items[0].metadata.name)
-  if [ "$VALIDATOR_POD" == "" ];
-  then
-    # Retry if an error occured
-    continue
-  fi
+oc wait --for=condition=Available --timeout=${timeout}s kubevirt/kubevirt -n $namespace
+
+oc apply -f - <<EOF
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kubevirt-config
+  namespace: kubevirt
+data:
+  feature-gates: "DataVolumes"
+---
+EOF
+
+key="/tmp/secrets/accessKeyId"
+token="/tmp/secrets/secretKey"
+
+if test -f "$key" && test -f "$token"; then
+  id=$(cat $key | tr -d '\n' | base64)
+  token=$(cat $token | tr -d '\n' | base64 | tr -d ' \n')
+
+  oc apply -n $namespace -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: common-templates-container-disk-puller
+  labels:
+    app: containerized-data-importer
+type: Opaque
+data:
+  accessKeyId: "${id}"
+  secretKey: "${token}"
+EOF
+fi
 
-  CA_BUNDLE=$(_oc exec -n $NAMESPACE $VALIDATOR_POD -- /bin/cat /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt | base64 -w 0)
-  if [ "$CA_BUNDLE" == "" ];
-  then
-    # Retry if an error occured
-    continue
+echo "Deploying CDI"
+export CDI_VERSION=$(curl -s https://api.github.com/repos/kubevirt/containerized-data-importer/releases | \
+            jq '.[] | select(.prerelease==false) | .tag_name' | sort -V | tail -n1 | tr -d '"')
+oc apply -f https://github.com/kubevirt/containerized-data-importer/releases/download/$CDI_VERSION/cdi-operator.yaml
+oc apply -f https://github.com/kubevirt/containerized-data-importer/releases/download/$CDI_VERSION/cdi-cr.yaml
+oc rollout status -n cdi deployment/cdi-operator
+oc apply -f - <<EOF
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: cdi-role
+  namespace: cdi
+rules:
+- apiGroups: ["cdi.kubevirt.io"]
+  resources: ["datavolumes/source"]
+  verbs: ["*"]
+---
+EOF
+
+export SSP_VERSION=$(curl -s https://api.github.com/repos/kubevirt/kubevirt-ssp-operator/releases | \
+            jq '.[] | select(.prerelease==false) | .tag_name' | sort -V | tail -n1 | tr -d '"')
+
+oc apply -n $namespace -f https://github.com/kubevirt/kubevirt-ssp-operator/releases/download/$SSP_VERSION/kubevirt-ssp-operator-crd.yaml
+oc apply -n $namespace -f https://github.com/kubevirt/kubevirt-ssp-operator/releases/download/$SSP_VERSION/kubevirt-ssp-operator.yaml
+
+oc apply -n $namespace -f - <<EOF
+apiVersion: ssp.kubevirt.io/v1
+kind: KubevirtTemplateValidator
+metadata:
+  name: kubevirt-template-validator
+  namespace: kubevirt
+spec:
+  templateValidatorReplicas: 2
+---
+EOF
+
+while [ $(oc get pods -n $namespace  | grep validator | wc -l) -eq 0 ] ; do 
```

Comment:
You should wait for the `Available` conditin on the validator deployment if you are waiting for it to be ready to accept requests